pkg/ebpf/c/tracee.bpf.c

```diff
@@ -5118,15 +5187,52 @@ int BPF_KPROBE(cgroup_bpf_run_filter_skb)
     if (!init_program_data(&p, ctx))
         return 0;
 
-    // obtain socket inode
+    bool mightbecloned = false; // cloned sock structs come from accept()
+
+    // obtain the socket inode using current "sock" structure
+
     u64 inode = BPF_READ(sk, sk_socket, file, f_inode, i_ino);
     if (inode == 0)
-        return 0; // e.g. vhost kernel threads might not have an inode
+        mightbecloned = true; // kernel threads might have zero inode
 
-    // pick network context from the inodemap (inode <=> task)
-    net_task_context_t *netctx = bpf_map_lookup_elem(&inodemap, &inode);
-    if (!netctx)
-        return 0; // e.g. task isn't being traced
+    struct net_task_context *netctx;
+
+    // obtain the task ctx using the obtained socket inode
+
+    if (!mightbecloned) {
+        // pick network context from the inodemap (inode <=> task)
+        netctx = bpf_map_lookup_elem(&inodemap, &inode);
+        if (!netctx)
+            mightbecloned = true; // e.g. task isn't being traced
```

Comment:
if we get here, is there a reason to continue looking for the `net_task_context`? 
correct me if i'm wrong, but if we had this info saved in the map it would be pointed by the inode we already got, isn't it?

Reply:
Not sure I follow your question :\. Lets see..

1. I check if the "sock" struct has an associated inode with it. For most of the cases, of existing sockets already (on going communication after socket was created) this is the case (no matter if code was called by either task or kernel context).

2. If inode is zero I know it "could be" a clone socket (or something else, sometimes, through observation).

3. IF the inode isn't zero, I try to obtain the task context from a saved inode <=> task relation. That is true for most of the cases actually (for existing/on-going flows from existing sockets). 

So, at (3) if I find the task context from the current inode, I' m good and will continue the existing logic (before this PR). Which is the default path and mechanism.

4. BUT if the inode was ZERO OR the task context could not be found (either because the task isn't being traced OR because this is a socket that was JUST created, by a kernel context, so no tracks exist)... then I need another check: check if I have this "sock" created by a "syn_recv_sock" execution path, or, in other words, check if this "unseen socket" came from the tcp handshake. IF IT CAME, then check if there is a task associated with it (which would be currently a task associated with the socket that created it). IF IT DID NOT COME from that path, then its safe to ignore the new socket (either unrelated OR not being traced).